src/System.Runtime.Extensions/src/System/IO/Path.WinRT.cs

```diff
@@ -11,12 +12,18 @@ namespace System.IO
 {
     public static partial class Path
     {
-        private static byte[] CreateCryptoRandomByteArray(int byteLength)
+        private static unsafe void GetCryptoRandomBytes(byte* bytes, int byteCount)
         {
+            Debug.Assert(bytes != null);
+            Debug.Assert(byteCount >= 0);
+
             byte[] arr;
-            IBuffer buffer = CryptographicBuffer.GenerateRandom((uint)byteLength);
+            IBuffer buffer = CryptographicBuffer.GenerateRandom((uint)byteCount);
             CryptographicBuffer.CopyToByteArray(buffer, out arr);
-            return arr;
+
+            Debug.Assert(arr.Length == byteCount);
+
+            Marshal.Copy(arr, 0, new IntPtr(bytes), byteCount);
```

Comment:
If you know of a more efficient way, let me know. I don't think it's a net regression due to the other `char[]` and `string` allocation reductions elsewhere and the small array size.

One idea I had was to query to see if the result of `CryptographicBuffer.GenerateRandom` implements `IBufferByteAccess` and copy the data directly instead of allocating a temp array via `CryptographicBuffer.CopyToByteArray`. However, I'm not familiar with building/testing the WinRT build of this library and I'm not sure if the result actually is an `IBufferByteAccess`. I suppose I could build a simple UWP app to try this out.